dev_guide/builds.adoc

```diff
@@ -846,42 +851,43 @@ source:
-binary as input, and an inline Dockerfile is optional.
+`*Binary*` is a valid value for `*BuildConfig.spec.source.type*`, but is it not something you will
+*typically* specify in any `*BuildConfig*` that you will author.
 
-The binary is generally assumed to be a tar, gzipped tar, or zip file depending
-on the strategy. For `*Docker*` builds, this is the build context and an
-optional Dockerfile may be specified to override any Dockerfile in the build
-context. For `*Source*` builds, this is assumed to be an archive as described
-above. For `*Source*` and `*Docker*` builds, if `*binary.asFile*` is set the
-build context will consist of a single file named by the value of `*binary.asFile*`. 
-The `*contextDir*` field may be used when an archive is provided. Custom builds will 
-receive this binary as input on standard input (`stdin`).
+This source type is unique in that it is leveraged based on your use of the `oc start-build` and
+`oc new-build` commands.
 
-A binary source potentially extracts content (if `*asFile*` is not set), in which case `*contextDir*`
-allows changing to a subdirectory within the content before the build executes.
+With `oc start-build` consider these options:
 
-The source definition is part of the `*spec*` section in the `*BuildConfig*`:
+1) `--from-file`:  The contents of the file you specify will be sent as a binary stream to the builder.  Then
+the builder will store the data in a file with the same name at the top of the build context (this internally maps
+to the `*binary.asFile*` field of the `*Binary*` source type).
 
-====
-[source,yaml]
-----
-source:
-  type: "Binary"
-  binary: <1>
-    asFile: "webapp.war" <2>
-  contextDir: "app/dir" <3>
-  dockerfile: "FROM centos:7\nRUN yum install -y httpd" <4>
-----
-<1> The `*binary*` field specifies the details of the binary source.
-<2> The `*asFile*` field specifies the name of the file that will be created with
-the binary contents.
-<3> The `*contextDir*` field specifies a subdirectory within the extracted contents of a
-binary archive.
-<4> If the optional `*dockerfile*` field is provided, it should be a string
-containing an inline Dockerfile that potentially replaces one within the
-contents of the binary archive.
-====
+2) `--from-dir` and `--from-repo`:  The contents will be archived (either zip, tar, or gzipped tar based on what
+is available on the client system) and sent as a binary stream to the builder.  Then the builder will extract
+the contents of the archive off of the top of the build context.
+
+In each of the above cases:
+
+1) If your `*BuildConfig*` already has a `*Binary*` source type defined, it will effectively be ignored and replaced by what the
+client sends.
+
+2) If your `*BuildConfig*` has a `*Git*` source type defined, it will be dynamically disabled, since `*Binary*` and `*Git*` are
+mutually exclusive, and the data in the binary stream provided to the builder will take precedence.
+
+3) Triggers will be ignored.
```

Comment:
Probably don't need to say this, if you're in a path were you're running "oc start-build" triggers aren't relevant.

but somewhere in here we should probably have a note that says something like "Note: binary type builds require content to be streamed from the local file system.  As such, it is not possible to automatically trigger a binary type build (eg via an image change trigger) because the binary files could not be provided.  Similarly you cannot launch binary type builds from the web console."